iocore/net/SSLUtils.cc

```diff
@@ -1743,11 +1827,70 @@ ssl_error_t
 SSLAccept(SSL *ssl)
 {
   ERR_clear_error();
-  int ret = SSL_accept(ssl);
+
+  int ret       = 0;
+  int ssl_error = SSL_ERROR_NONE;
+
+#if TS_HAS_TLS_EARLY_DATA
+  if (SSL_version(ssl) < TLS1_3_VERSION) {
+    ret = SSL_accept(ssl);
+  } else {
+    SSLNetVConnection *netvc = SSLNetVCAccess(ssl);
+
+    if (SSLConfigParams::server_max_early_data > 0 && !netvc->early_data_finish) {
+      size_t nread;
+      if (netvc->early_data_buf == nullptr) {
+        netvc->early_data_buf    = new_MIOBuffer(BUFFER_SIZE_INDEX_16K);
+        netvc->early_data_reader = netvc->early_data_buf->alloc_reader();
+      }
+
+      while (true) {
+        IOBufferBlock *block = new_IOBufferBlock();
+        block->alloc(BUFFER_SIZE_INDEX_16K);
+        ret = SSL_read_early_data(ssl, block->buf(), index_to_buffer_size(BUFFER_SIZE_INDEX_16K), &nread);
+
+        if (ret == SSL_READ_EARLY_DATA_ERROR) {
+          Debug("ssl_early_data", "SSL_READ_EARLY_DATA_ERROR");
+          break;
+        } else {
+          if (nread > 0) {
+            block->fill(nread);
+            netvc->early_data_buf->append_block(block);
+            SSL_INCREMENT_DYN_STAT(ssl_early_data_received_count);
+
+            if (is_debug_tag_set("ssl_early_data_show_received")) {
+              std::string early_data_str(reinterpret_cast<char *>(block->buf()), nread);
+              Debug("ssl_early_data_show_received", "Early data buffer: \n%s", early_data_str.c_str());
```

Comment:
I want to have "v_" prefix for the debug tag, because current tag matches when you specify just "ssl" to `debug.tags`. I think it's a bit too verbose.

Reply:
Ah, I missed `is_debug_tag_set("ssl_early_data_show_received")`.